blog-server-services/src/impls/rbatis_author_service.rs

```diff
@@ -7,9 +7,14 @@ pub fn create_rbatis_author_service(rb: RBatis) -> Box<dyn AuthorService> {
 }
 
 impl_insert!(BaseAuthor {}, "author");
-impl_select!(Author {select_by_id(id: &i64) -> Option => "`WHERE id = #{id} LIMIT 1`"});
-impl_select!(Author {select_by_slug(slug: &String) -> Option => "`WHERE slug = #{slug} LIMIT 1`"});
-impl_select!(Author {select_all_with_offset_and_limit(offset: &i64, limit: &i64) => "`LIMIT #{limit} OFFSET #{offset}`"});
+impl_select!(Author {select_by_id(id: &i64) -> Option => 
+    "`WHERE id = #{id} LIMIT 1`"});
+impl_select!(Author {select_by_slug(slug: &String) -> Option => 
+    "`WHERE slug = #{slug} LIMIT 1`"});
+impl_select!(Author {select_all_with_offset_and_limit(offset: &i64, limit: &i64) => 
+    "`LIMIT #{limit} OFFSET #{offset}`"});
+impl_select!(Author {select_all_by_query_with_offset_and_limit(query: &String, offset: &i64, limit: &i64) => 
+    "`WHERE author.slug ILIKE '%' || #{query} || '%' OR author.first_name ILIKE '%' || #{query} || '%' OR author.middle_name ILIKE '%' || #{query} || '%' OR author.last_name ILIKE '%' || #{query} || '%' LIMIT #{limit} OFFSET #{offset}`"});
```

Comment:
There is a possibility that 2 users will change let say their middlenames after our first pagination call and before second, so each time we will obtain different set of data. 

i dont think we have workaroung for now and for MVP could be ok, but maybe we should take it in consideration in the future.

Also im not sure that order of records will be consistent

Reply:
- "There is a possibility that 2 users will change let say their middlenames after our first pagination call and before second, so each time we will obtain different set of data."
**It's expected I believe, same situation for all other places with pagination (posts moved on other pages and etc.)**
- i dont think we have workaroung for now and for MVP could be ok, but maybe we should take it in consideration in the future.
**Will create task.**
- Also im not sure that order of records will be consistent
Good point, will check.

Reply:
- [ ] Check "Also im not sure that order of records will be consistent"

Reply:
Looks sorted as placed in database, for now applicable.